src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1351,9 +1402,60 @@ private void getVars(CORINFO_METHOD_STRUCT_* ftn, ref uint cVars, ILVarInfo** va
             // Just tell the JIT to extend everything.
             extendOthers = true;
         }
+
         private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
         {
-            // TODO: Debugging
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
+            {
+                return;
+            }
+
+            Dictionary<uint, DebugVarInfo> debugVars = new Dictionary<uint, DebugVarInfo>();
+            int i;
+
+            for (i = 0; i < cVars; i++)
+            {
+                NativeVarInfo nativeVarInfo = vars[i];
+
+                if (nativeVarInfo.varNumber < _parameterIndexToNameMap.Count)
+                {
+                    string name = _parameterIndexToNameMap[nativeVarInfo.varNumber];
+                    updateDebugVarInfo(debugVars, nativeVarInfo.varNumber, name, true, nativeVarInfo);
+                }
+                else
+                {
+                    ILLocalVariable ilVar;
+                    uint slotNumber = nativeVarInfo.varNumber - (uint)_parameterIndexToNameMap.Count;
+                    if (_localSlotToInfoMap.TryGetValue(slotNumber, out ilVar))
+                    {
+                        updateDebugVarInfo(debugVars, nativeVarInfo.varNumber, ilVar.Name, true, nativeVarInfo);
```

Comment:
Or even make it an enum with two values instead of the bool so that it's easy to see what it means. It's too easy to mess this up if it's a bool...